From 89ff9d58e6b61bc90638157ae9ec21f6e3186a5a Mon Sep 17 00:00:00 2001
From: Arnd Bergmann <arnd@arndb.de>
Date: Wed, 23 Aug 2017 15:22:52 +0200
Subject: [PATCH] staging: rtlwifi: add MAC80211 dependency
Git-commit: 89ff9d58e6b61bc90638157ae9ec21f6e3186a5a
Patch-mainline: v4.14-rc1
References: FATE#326887

Like the version in drivers/net/wireless, this driver requires the
MAC80211 framework, otherwise we run into a link error:

Error: "ieee80211_rx_irqsafe" [drivers/staging/rtlwifi/r8822be.ko] undefined!
Error: "cfg80211_unlink_bss" [drivers/staging/rtlwifi/r8822be.ko] undefined!
Error: "ieee80211_beacon_get_tim" [drivers/staging/rtlwifi/r8822be.ko] undefined!
Error: "ieee80211_resume_disconnect" [drivers/staging/rtlwifi/r8822be.ko] undefined!

This adds the Kconfig dependency for it.

Fixes: 56bde846304e ("staging: r8822be: Add existing rtlwifi and rtl_pci parts for new driver")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/staging/rtlwifi/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/rtlwifi/Kconfig b/drivers/staging/rtlwifi/Kconfig
index 37e47b80594a..cb3a29ae764b 100644
--- a/drivers/staging/rtlwifi/Kconfig
+++ b/drivers/staging/rtlwifi/Kconfig
@@ -1,6 +1,6 @@
 config R8822BE
 	tristate "Realtek RTL8822BE Wireless Network Adapter"
-	depends on NETDEVICES && PCI && m
+	depends on PCI && MAC80211 && m
 	select FW_LOADER
 	---help---
 	This is the staging driver for Realtek RTL8822BE 802.11ac PCIe
-- 
2.19.1

